File: Alvas.Audio/AudioExVB/VoxPanel.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Text
Imports System.Windows.Forms
Imports System.IO
Imports Alvas.Audio
Imports System.Diagnostics

Namespace AudioExCS
	Public Class VoxPanel
		Inherits UserControl
        Private ofdVox As System.Windows.Forms.OpenFileDialog
        Private WithEvents btnConvert As Button
        Private WithEvents btnOpen As Button
		Private cbSamplesPerSec As ComboBox
		Private tbVox As TextBox
        Private WithEvents btnConvert2 As Button
        Private WithEvents btnWav As Button
		Private cbSamplesPerSec2 As ComboBox
		Private tbWav As TextBox
        Private WithEvents cbResample As CheckBox
        Private WithEvents btnVox2Mp3 As System.Windows.Forms.Button

        ''' <summary> 
        ''' Required designer variable.
        ''' </summary>
        Private components As System.ComponentModel.IContainer = Nothing

        ''' <summary> 
        ''' Clean up any resources being used.
        ''' </summary>
        ''' <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing AndAlso Not (components Is Nothing) Then
                components.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub

#Region "Component Designer generated code"

        ''' <summary> 
        ''' Required method for Designer support - do not modify 
        ''' the contents of this method with the code editor.
        ''' </summary>
        Private Sub InitializeComponent()
            Me.btnConvert = New System.Windows.Forms.Button
            Me.btnOpen = New System.Windows.Forms.Button
            Me.cbSamplesPerSec = New System.Windows.Forms.ComboBox
            Me.tbVox = New System.Windows.Forms.TextBox
            Me.btnConvert2 = New System.Windows.Forms.Button
            Me.btnWav = New System.Windows.Forms.Button
            Me.cbSamplesPerSec2 = New System.Windows.Forms.ComboBox
            Me.tbWav = New System.Windows.Forms.TextBox
            Me.cbResample = New System.Windows.Forms.CheckBox
            Me.btnVox2Mp3 = New System.Windows.Forms.Button
            ofdVox = New System.Windows.Forms.OpenFileDialog
            Me.SuspendLayout()
            '
            'btnConvert
            '
            Me.btnConvert.Location = New System.Drawing.Point(6, 58)
            Me.btnConvert.Name = "btnConvert"
            Me.btnConvert.Size = New System.Drawing.Size(212, 23)
            Me.btnConvert.TabIndex = 5
            Me.btnConvert.Text = "Vox -> Wav"
            '
            'btnOpen
            '
            Me.btnOpen.Location = New System.Drawing.Point(194, 2)
            Me.btnOpen.Name = "btnOpen"
            Me.btnOpen.Size = New System.Drawing.Size(24, 23)
            Me.btnOpen.TabIndex = 4
            Me.btnOpen.Text = "..."
            '
            'cbSamplesPerSec
            '
            Me.cbSamplesPerSec.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
            Me.cbSamplesPerSec.Location = New System.Drawing.Point(6, 31)
            Me.cbSamplesPerSec.Name = "cbSamplesPerSec"
            Me.cbSamplesPerSec.Size = New System.Drawing.Size(212, 21)
            Me.cbSamplesPerSec.TabIndex = 7
            '
            'tbVox
            '
            Me.tbVox.Location = New System.Drawing.Point(6, 5)
            Me.tbVox.Name = "tbVox"
            Me.tbVox.Size = New System.Drawing.Size(187, 20)
            Me.tbVox.TabIndex = 6
            '
            'btnConvert2
            '
            Me.btnConvert2.Location = New System.Drawing.Point(6, 196)
            Me.btnConvert2.Name = "btnConvert2"
            Me.btnConvert2.Size = New System.Drawing.Size(212, 23)
            Me.btnConvert2.TabIndex = 9
            Me.btnConvert2.Text = "Wav -> Vox"
            '
            'btnWav
            '
            Me.btnWav.Location = New System.Drawing.Point(194, 117)
            Me.btnWav.Name = "btnWav"
            Me.btnWav.Size = New System.Drawing.Size(24, 23)
            Me.btnWav.TabIndex = 8
            Me.btnWav.Text = "..."
            '
            'cbSamplesPerSec2
            '
            Me.cbSamplesPerSec2.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
            Me.cbSamplesPerSec2.Enabled = False
            Me.cbSamplesPerSec2.Location = New System.Drawing.Point(6, 169)
            Me.cbSamplesPerSec2.Name = "cbSamplesPerSec2"
            Me.cbSamplesPerSec2.Size = New System.Drawing.Size(212, 21)
            Me.cbSamplesPerSec2.TabIndex = 11
            '
            'tbWav
            '
            Me.tbWav.Location = New System.Drawing.Point(6, 120)
            Me.tbWav.Name = "tbWav"
            Me.tbWav.Size = New System.Drawing.Size(187, 20)
            Me.tbWav.TabIndex = 10
            '
            'cbResample
            '
            Me.cbResample.Location = New System.Drawing.Point(6, 146)
            Me.cbResample.Name = "cbResample"
            Me.cbResample.Size = New System.Drawing.Size(90, 17)
            Me.cbResample.TabIndex = 12
            Me.cbResample.Text = "Resample"
            '
            'ofdVox
            '
            ofdVox.Filter = "*.vox|*.vox"
            '
            'btnVox2Mp3
            '
            Me.btnVox2Mp3.Location = New System.Drawing.Point(6, 87)
            Me.btnVox2Mp3.Name = "btnVox2Mp3"
            Me.btnVox2Mp3.Size = New System.Drawing.Size(212, 23)
            Me.btnVox2Mp3.TabIndex = 14
            Me.btnVox2Mp3.Text = "Vox -> Mp3"
            '
            'VoxPanel
            '
            Me.Controls.Add(Me.btnVox2Mp3)
            Me.Controls.Add(Me.cbResample)
            Me.Controls.Add(Me.btnConvert2)
            Me.Controls.Add(Me.btnWav)
            Me.Controls.Add(Me.cbSamplesPerSec2)
            Me.Controls.Add(Me.tbWav)
            Me.Controls.Add(Me.btnConvert)
            Me.Controls.Add(Me.btnOpen)
            Me.Controls.Add(Me.cbSamplesPerSec)
            Me.Controls.Add(Me.tbVox)
            Me.Name = "VoxPanel"
            Me.Size = New System.Drawing.Size(224, 243)
            Me.ResumeLayout(False)
            Me.PerformLayout()

        End Sub

#End Region

        Public Sub New()
            InitializeComponent()
            cbSamplesPerSec.DataSource = New Integer() {6000, 8000, 11025, 12000, 16000, 22050, _
             24000, 32000, 44100, 48000}
            cbSamplesPerSec2.DataSource = New Integer() {6000, 8000, 11025, 12000, 16000, 22050, _
             24000, 32000, 44100, 48000}
        End Sub

        Private Sub btnConvert_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnConvert.Click
            Dim voxFile As String = tbVox.Text
            If voxFile = String.Empty Then
                MessageBox.Show("Please, select file for converting")
                Return
            End If
            Dim wavFile As String = Path.ChangeExtension(voxFile, ".wav")
            Vox2Wav(voxFile, wavFile, CInt(cbSamplesPerSec.SelectedValue))
            MessageBox.Show(String.Format("The result is stored in the file '{0}'. Choose next file for the converting.", wavFile), "The conversion was executed successfully")
            OpenContainingFolder(wavFile)
        End Sub

        Private Sub OpenContainingFolder(ByVal fileName As String)
            Process.Start(Path.GetDirectoryName(fileName))
        End Sub

        Private Shared Sub Vox2Wav(ByVal voxFile As String, ByVal wavFile As String, ByVal samplesPerSec As Integer)
            Dim br As New BinaryReader(File.OpenRead(voxFile))
            Dim format As IntPtr = AudioCompressionManager.GetPcmFormat(1, 16, samplesPerSec)
            Dim ww As New WaveWriter(File.Create(wavFile), AudioCompressionManager.FormatBytes(format))
            Vox.Vox2Wav(br, ww)
            br.Close()
            ww.Close()
        End Sub

        Private Sub btnOpen_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnOpen.Click
            OpenFile(VoxFilter, tbVox)
        End Sub

        Private Sub btnWav_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnWav.Click
            OpenFile(WavFilter, tbWav)
        End Sub

        Private Sub OpenFile(ByVal filter As String, ByVal tb As TextBox)
            ofdVox.Filter = filter
            If ofdVox.ShowDialog() = DialogResult.OK Then
                tb.Text = ofdVox.FileName
            End If
        End Sub

        Private Const VoxFilter As String = "*.vox|*.vox"
        Private Const WavFilter As String = "*.wav|*.wav"

        Private Sub btnConvert2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnConvert2.Click
            Dim wavFile As String = tbWav.Text
            If wavFile = String.Empty Then
                MessageBox.Show("Please, select file for converting")
                Return
            End If
            Dim voxFile As String = Path.ChangeExtension(wavFile, ".vox")
            Wav2Vox(wavFile, voxFile, CInt(cbSamplesPerSec2.SelectedValue), cbResample.Checked)
            MessageBox.Show(String.Format("The result is stored in the file '{0}'. Choose next file for the converting.", voxFile), "The conversion was executed successfully")
            OpenContainingFolder(voxFile)
        End Sub

        Private Shared Sub Wav2Vox(ByVal inFile As String, ByVal outFile As String, ByVal samplesPerSec As Integer, ByVal isResample As Boolean)
            Dim wr As New WaveReader(File.OpenRead(inFile))
            Dim format As IntPtr = wr.ReadFormat()
            Dim data As Byte() = wr.ReadData()
            wr.Close()
            Dim wf As WaveFormat = AudioCompressionManager.GetWaveFormat(format)
            If wf.wFormatTag <> AudioCompressionManager.PcmFormatTag Then
                'Decode if not PCM data 
                Decode2Pcm(format, data, wf)
            End If
            If isResample AndAlso wf.nSamplesPerSec <> samplesPerSec Then
                Resample(format, data, wf, samplesPerSec)
            End If
            Dim bw As New BinaryWriter(File.OpenWrite(outFile))
            Dim br As New BinaryReader(New MemoryStream(data))
            Vox.Raw2Vox(br, bw, wf.wBitsPerSample)
            br.Close()
            bw.Close()
        End Sub

        Private Shared Sub Resample(ByRef format As IntPtr, ByRef data As Byte(), ByRef wf As WaveFormat, ByVal samplesPerSec As Integer)
            Dim newFormat As IntPtr = AudioCompressionManager.GetPcmFormat(wf.nChannels, wf.wBitsPerSample, samplesPerSec)
            Dim buffer As Byte() = AudioCompressionManager.Convert(format, newFormat, data, True)
            format = newFormat
            wf = AudioCompressionManager.GetWaveFormat(newFormat)
            data = buffer
        End Sub

        Private Shared Sub Decode2Pcm(ByRef format As IntPtr, ByRef data As Byte(), ByRef wf As WaveFormat)
            Dim newFormat As IntPtr = AudioCompressionManager.GetCompatibleFormat(format, AudioCompressionManager.PcmFormatTag)
            Dim buffer As Byte() = AudioCompressionManager.Convert(format, newFormat, data, False)
            wf = AudioCompressionManager.GetWaveFormat(newFormat)
            format = newFormat
            data = buffer
        End Sub

        Private Sub cbResample_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbResample.CheckedChanged
            cbSamplesPerSec2.Enabled = cbResample.Checked
        End Sub

        Private Sub btnVox2Mp3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnVox2Mp3.Click
            Dim voxFile As String = tbVox.Text
            If voxFile = String.Empty Then
                MessageBox.Show("Please, select file for converting")
                Return
            End If
            Dim mp3File As String = Path.ChangeExtension(voxFile, ".mp3")
            Vox.Vox2Mp3(voxFile, mp3File, CInt(cbSamplesPerSec.SelectedValue))
            MessageBox.Show(String.Format("The result is stored in the file '{0}'. Choose next file for the converting.", mp3File), "The conversion was executed successfully")
            OpenContainingFolder(mp3File)
        End Sub
    End Class
End Namespace

File: Alvas.Audio/AudioExCS/VoxPanel.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using System.IO;
using Alvas.Audio;
using System.Diagnostics;

namespace AudioExCS
{
    public class VoxPanel : UserControl
    {
        private OpenFileDialog ofdVox;
        private Button btnConvert;
        private Button btnOpen;
        private ComboBox cbSamplesPerSec;
        private TextBox tbVox;
        private Button btnConvert2;
        private Button btnWav;
        private ComboBox cbSamplesPerSec2;
        private TextBox tbWav;
        private CheckBox cbResample;
        private Button btnVox2Mp3;

        /// <summary> 
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.IContainer components = null;

        /// <summary> 
        /// Clean up any resources being used.
        /// </summary>
        /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
        protected override void Dispose(bool disposing)
        {
            if (disposing && (components != null))
            {
                components.Dispose();
            }
            base.Dispose(disposing);
        }

        #region Component Designer generated code

        /// <summary> 
        /// Required method for Designer support - do not modify 
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.btnConvert = new System.Windows.Forms.Button();
            this.btnOpen = new System.Windows.Forms.Button();
            this.cbSamplesPerSec = new System.Windows.Forms.ComboBox();
            this.tbVox = new System.Windows.Forms.TextBox();
            this.btnConvert2 = new System.Windows.Forms.Button();
            this.btnWav = new System.Windows.Forms.Button();
            this.cbSamplesPerSec2 = new System.Windows.Forms.ComboBox();
            this.tbWav = new System.Windows.Forms.TextBox();
            this.cbResample = new System.Windows.Forms.CheckBox();
            this.btnVox2Mp3 = new System.Windows.Forms.Button();
            this.ofdVox = new System.Windows.Forms.OpenFileDialog();
            this.SuspendLayout();
            // 
            // btnConvert
            // 
            this.btnConvert.Location = new System.Drawing.Point(6, 58);
            this.btnConvert.Name = "btnConvert";
            this.btnConvert.Size = new System.Drawing.Size(212, 23);
            this.btnConvert.TabIndex = 5;
            this.btnConvert.Text = "Vox -> Wav";
            this.btnConvert.Click += new System.EventHandler(this.btnConvert_Click);
            // 
            // btnOpen
            // 
            this.btnOpen.Location = new System.Drawing.Point(194, 2);
            this.btnOpen.Name = "btnOpen";
            this.btnOpen.Size = new System.Drawing.Size(24, 23);
            this.btnOpen.TabIndex = 4;
            this.btnOpen.Text = "...";
            this.btnOpen.Click += new System.EventHandler(this.btnOpen_Click);
            // 
            // cbSamplesPerSec
            // 
            this.cbSamplesPerSec.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
            this.cbSamplesPerSec.Location = new System.Drawing.Point(6, 31);
            this.cbSamplesPerSec.Name = "cbSamplesPerSec";
            this.cbSamplesPerSec.Size = new System.Drawing.Size(212, 21);
            this.cbSamplesPerSec.TabIndex = 7;
            // 
            // tbVox
            // 
            this.tbVox.Location = new System.Drawing.Point(6, 5);
            this.tbVox.Name = "tbVox";
            this.tbVox.Size = new System.Drawing.Size(187, 20);
            this.tbVox.TabIndex = 6;
            // 
            // btnConvert2
            // 
            this.btnConvert2.Location = new System.Drawing.Point(6, 196);
            this.btnConvert2.Name = "btnConvert2";
            this.btnConvert2.Size = new System.Drawing.Size(212, 23);
            this.btnConvert2.TabIndex = 9;
            this.btnConvert2.Text = "Wav -> Vox";
            this.btnConvert2.Click += new System.EventHandler(this.btnConvert2_Click);
            // 
            // btnWav
            // 
            this.btnWav.Location = new System.Drawing.Point(194, 117);
            this.btnWav.Name = "btnWav";
            this.btnWav.Size = new System.Drawing.Size(24, 23);
            this.btnWav.TabIndex = 8;
            this.btnWav.Text = "...";
            this.btnWav.Click += new System.EventHandler(this.btnWav_Click);
            // 
            // cbSamplesPerSec2
            // 
            this.cbSamplesPerSec2.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
            this.cbSamplesPerSec2.Enabled = false;
            this.cbSamplesPerSec2.Location = new System.Drawing.Point(6, 169);
            this.cbSamplesPerSec2.Name = "cbSamplesPerSec2";
            this.cbSamplesPerSec2.Size = new System.Drawing.Size(212, 21);
            this.cbSamplesPerSec2.TabIndex = 11;
            // 
            // tbWav
            // 
            this.tbWav.Location = new System.Drawing.Point(6, 120);
            this.tbWav.Name = "tbWav";
            this.tbWav.Size = new System.Drawing.Size(187, 20);
            this.tbWav.TabIndex = 10;
            // 
            // cbResample
            // 
            this.cbResample.Location = new System.Drawing.Point(6, 146);
            this.cbResample.Name = "cbResample";
            this.cbResample.Size = new System.Drawing.Size(90, 17);
            this.cbResample.TabIndex = 12;
            this.cbResample.Text = "Resample";
            this.cbResample.CheckedChanged += new System.EventHandler(this.cbResample_CheckedChanged);
            // 
            // ofdVox
            // 
            ofdVox.Filter = "*.vox|*.vox";
            // 
            // btnVox2Mp3
            // 
            this.btnVox2Mp3.Location = new System.Drawing.Point(6, 87);
            this.btnVox2Mp3.Name = "btnVox2Mp3";
            this.btnVox2Mp3.Size = new System.Drawing.Size(212, 23);
            this.btnVox2Mp3.TabIndex = 13;
            this.btnVox2Mp3.Text = "Vox -> Mp3";
            this.btnVox2Mp3.Click += new System.EventHandler(this.btnVox2Mp3_Click);
            // 
            // VoxPanel
            // 
            this.Controls.Add(this.btnVox2Mp3);
            this.Controls.Add(this.cbResample);
            this.Controls.Add(this.btnConvert2);
            this.Controls.Add(this.btnWav);
            this.Controls.Add(this.cbSamplesPerSec2);
            this.Controls.Add(this.tbWav);
            this.Controls.Add(this.btnConvert);
            this.Controls.Add(this.btnOpen);
            this.Controls.Add(this.cbSamplesPerSec);
            this.Controls.Add(this.tbVox);
            this.Name = "VoxPanel";
            this.Size = new System.Drawing.Size(224, 243);
            this.ResumeLayout(false);
            this.PerformLayout();

        }

        #endregion

        public VoxPanel()
        {
            InitializeComponent();
            cbSamplesPerSec.DataSource = new int[] {6000, 8000, 11025, 12000, 16000,
22050, 24000, 32000, 44100, 48000};
            cbSamplesPerSec2.DataSource = new int[] {6000, 8000, 11025, 12000, 16000,
22050, 24000, 32000, 44100, 48000};
        }

        private void btnConvert_Click(object sender, EventArgs e)
        {
            string voxFile = tbVox.Text;
            if (voxFile == string.Empty)
            {
                MessageBox.Show("Please, select file for converting");
                return;
            }
            string wavFile = Path.ChangeExtension(voxFile, ".wav");
            Vox2Wav(voxFile, wavFile, (int)cbSamplesPerSec.SelectedValue);
            MessageBox.Show(string.Format("The result is stored in the file '{0}'. Choose next file for the converting.", wavFile), "The conversion was executed successfully");
            OpenContainingFolder(wavFile);
        }

        private void OpenContainingFolder(string fileName)
        {
            Process.Start(Path.GetDirectoryName(fileName));
        }

        private static void Vox2Wav(string voxFile, string wavFile, int samplesPerSec)
        {
            BinaryReader br = new BinaryReader(File.OpenRead(voxFile));
            IntPtr format = AudioCompressionManager.GetPcmFormat(1, 16, samplesPerSec);
            WaveWriter ww = new WaveWriter(File.Create(wavFile), AudioCompressionManager.FormatBytes(format));
            Vox.Vox2Wav(br, ww);
            br.Close();
            ww.Close();
        }

        private void btnOpen_Click(object sender, EventArgs e)
        {
            OpenFile(VoxFilter, tbVox);
        }

        private void btnWav_Click(object sender, EventArgs e)
        {
            OpenFile(WavFilter, tbWav);
        }

        private void OpenFile(string filter, TextBox tb)
        {
            ofdVox.Filter = filter;
            if (ofdVox.ShowDialog() == DialogResult.OK)
            {
                tb.Text = ofdVox.FileName;
            }
        }

        private const string VoxFilter = "*.vox|*.vox";
        private const string WavFilter = "*.wav|*.wav";

        private void btnConvert2_Click(object sender, EventArgs e)
        {
            string wavFile = tbWav.Text;
            if (wavFile == string.Empty)
            {
                MessageBox.Show("Please, select file for converting");
                return;
            }
            string voxFile = Path.ChangeExtension(wavFile, ".vox");
            Wav2Vox(wavFile, voxFile, (int)cbSamplesPerSec2.SelectedValue, cbResample.Checked);
            MessageBox.Show(string.Format("The result is stored in the file '{0}'. Choose next file for the converting.", voxFile), "The conversion was executed successfully");
            OpenContainingFolder(voxFile);
        }

        private static void Wav2Vox(string inFile, string outFile, int samplesPerSec, bool isResample)
        {
            WaveReader wr = new WaveReader(File.OpenRead(inFile));
            IntPtr format = wr.ReadFormat();
            byte[] data = wr.ReadData();
            wr.Close();
            WaveFormat wf = AudioCompressionManager.GetWaveFormat(format);
            if (wf.wFormatTag != AudioCompressionManager.PcmFormatTag)//Decode if not PCM data 
            {
                Decode2Pcm(ref format, ref data, ref wf);
            }
            if (isResample && wf.nSamplesPerSec != samplesPerSec)
            {
                Resample(ref format, ref data, ref wf, samplesPerSec);
            }
            BinaryWriter bw = new BinaryWriter(File.OpenWrite(outFile));
            BinaryReader br = new BinaryReader(new MemoryStream(data));
            Vox.Raw2Vox(br, bw, wf.wBitsPerSample);
            br.Close();
            bw.Close();
        }

        private static void Resample(ref IntPtr format, ref byte[] data, ref WaveFormat wf, int samplesPerSec)
        {
            IntPtr newFormat = AudioCompressionManager.GetPcmFormat(wf.nChannels, wf.wBitsPerSample, samplesPerSec);
            byte[] buffer = AudioCompressionManager.Convert(format, newFormat, data, false);
            format = newFormat;
            wf = AudioCompressionManager.GetWaveFormat(newFormat);
            data = buffer;
        }

        private static void Decode2Pcm(ref IntPtr format, ref byte[] data, ref WaveFormat wf)
        {
            IntPtr newFormat = AudioCompressionManager.GetCompatibleFormat(format, AudioCompressionManager.PcmFormatTag);
            byte[] buffer = AudioCompressionManager.Convert(format, newFormat, data, false);
            wf = AudioCompressionManager.GetWaveFormat(newFormat);
            format = newFormat;
            data = buffer;
        }

        private void cbResample_CheckedChanged(object sender, EventArgs e)
        {
            cbSamplesPerSec2.Enabled = cbResample.Checked;
        }

        private void btnVox2Mp3_Click(object sender, EventArgs e)
        {
            string voxFile = tbVox.Text;
            if (voxFile == string.Empty)
            {
                MessageBox.Show("Please, select file for converting");
                return;
            }
            string mp3File = Path.ChangeExtension(voxFile, ".mp3");
            Vox.Vox2Mp3(voxFile, mp3File, (int)cbSamplesPerSec.SelectedValue);
            MessageBox.Show(string.Format("The result is stored in the file '{0}'. Choose next file for the converting.", mp3File), "The conversion was executed successfully");
            OpenContainingFolder(mp3File);
        }

    }
}
